lib/datadog/statsd/sender.rb

```diff
@@ -32,40 +46,56 @@ def rendez_vous
       def add(message)
         raise ArgumentError, 'Start sender first' unless message_queue
 
+        # if the thread does not exist, we assume we are running in a forked process,
+        # empty the message queue and message buffers (these messages belong to
+        # the parent process) and spawn a new companion thread.
+        if !sender_thread.alive?
+          @logger.warn { "Statsd: companion thread dead, re-creating one" } if @logger
```

Comment:
Same note as above: if a user sees this in their log, I'm not entirely sure they'll understand the meaning, and if it's anything they can fix or not. I suggest clarifying this from that POV.